test/unit/mem/buf/src/BufTest.cpp

```diff
@@ -15,6 +15,54 @@
 #include <numeric>
 #include <type_traits>
 
+namespace buftest
+{
+    template<typename TDim, typename TDev, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(TDev dev, TExtent extent)
+    {
+        return alpaka::allocBuf<TElem, TIdx>(dev, extent);
+    }
+
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevCpu dev, TExtent extent) -> alpaka::BufCpu<TElem, TDim, TIdx>
+    {
+        return alpaka::allocBuf<TElem, TIdx>(dev, extent);
+    }
+
+#if defined(ALPAKA_ACC_GPU_CUDA_ENABLED)
+    template<typename TDim, typename TElem, typename TIdx, typename TExtent>
+    auto allocBuf(alpaka::DevCudaRt dev, TExtent extent) -> alpaka::BufCudaRt<TElem, TDim, TIdx>
+    {
+        return alpaka::allocBuf<TElem, TIdx>(dev, extent);
+    }
+#elif defined(ALPAKA_ACC_GPU_HIP_ENABLED)
```

Comment:
I would change 
```suggestion
#endif

#if defined(ALPAKA_ACC_GPU_HIP_ENABLED)
```

so if a test enables multiple back-ends they can all be tested.

Reply:
Ok, done!